### PR TITLE
fix: find right config if same dbt test is added with diff configs in yml

### PR DESCRIPTION
**Pull request by saravmajestic**
### Problem
If a column has same type of generic test with diff config, doc editor does not save properly

### Solution
- While finding the config of the test from existing schema, equate the values to get right config

### Screenshot/Demo

A picture is worth a thousand words. Please highlight the changes if applicable.

### How to test
- add test as in issue #1219 for any column in dbt project
- change some description of model from doc editor
- save
- should save with right config

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -256,3 +260,51 @@ export const getExternalProjectNamesFromDbtLoomConfig = (
   }
   return null;
 };
+
+export const isRelationship = (
```
Comment (saravmajestic) [inline src/utils.ts:264]: extracted these utilities to separate file from docEditPanel.ts

File: src/utils.ts [hunk 2]
```diff
@@ -256,3 +260,51 @@ export const getExternalProjectNamesFromDbtLoomConfig = (
...
+
+    if (isAcceptedValues(kwargs)) {
+      return (
+        kwargs.values?.sort().toString() ===
+        t.accepted_values.values.sort().toString()
+      );
+    }
+
+    return true;
+  });
```
Comment (saravmajestic) [inline src/utils.ts:309]: this is the actual fix, which finds the config by values/field and to for generic tests

File: src/services/dbtTestService.ts [hunk 3]
```diff
@@ -25,10 +29,10 @@ export class DbtTestService {
   ) {}
 
   private filterAndStringifyTest = (
-    tests: Record<string, Record<string, unknown>>[],
+    testsPerColumnOrModelFromYml: Record<string, Record<string, unknown>>[],
```
Comment (saravmajestic) [inline src/services/dbtTestService.ts:32]: renamed variables

File: src/services/dbtTestService.ts [hunk 4]
```diff
@@ -41,25 +45,25 @@ export class DbtTestService {
     const { name, namespace } = test.test_metadata;
     const fullName = namespace ? `${namespace}.${name}` : name;
 
-    const selectedTest = tests.find(
-      (t: Record<string, Record<string, unknown>>) => {
-        return Boolean(t[fullName]);
-      },
+    const existingConfig = getColumnTestConfigFromYml(
```
Comment (saravmajestic) [inline src/services/dbtTestService.ts:48]: use the common utility method to get the right config

**Review by mdesmet — APPROVED** [review]

**Pull request merged**